### ISSUE TITLE
Translation API for Plugins

### ISSUE THREAD

**Issue by m1well** [NotePlan/plugins#135]
**Is your feature request related to a problem? Please describe.**
Currently I am extending the `weather.js` for multiple language support. But this is very hard.  
For the sake of simplicity I avoid a translation for the weather symbols or the `feels like` term.

**Describe the solution you'd like**
currently I can think of 3 ways to provide translations:
* translations only in the code
   * but they are tracked via git)
* translations in a new `_configuration` file named `_translations`
   * with json data structure
   * aren't tracked via git - so we have to provide a default file and add new translations over time 
* do a http fetch call on each plugin which needs translations (like the call to the config file)
   * we need an http api
   * maybe we can add a file here in the main branch and call this via `fetch`
   * then it is tracked via git

**Additional context**
file could possible look like this:
```json
  {  
    "en" : {
      "weather": {
        "starter": "Weather",
        "feelslike": "Feels like",
        "icons": {
          "showers": "showers",
          "rain": "rain",
          ...
        }
      }
    }
    "de" : {
      "weather": {
        "starter": "Wetter",
        "feelslike": "Gefühlt",
        "icons": {
          "showers": "Schauer",
          "rain": "Regen",
          ...
        }
      }
    }

  }
```

**Comment by mikeerickson** [comment]
@m1well In theory, I agree there needs to be better language support for all plugins, and is something that I have taken great care for in the templating plugin.  

In my case as simple example, for `DateModule` library, I am retrieving the users language from operating system so that it happens automatically, no additional configuration is necessary.

Further, the `weather` method in templating does not require any “configuration” as does the current `weather.js` implementation by @jgclark which also handles language automatically, as it uses a different weather service.

And finally, I am not keen to extending the current `_configuration` system we have right now as it is going through a refactor/upgrade using the new plugin settings interface currently in development, and will be available in the very near future.

In summary:

- plugins should use the operating system to determine current language (this will be documented separately when templating libraries are released). 
- All `_configuration` will become deprecated when NotePlan plugin settings interface is released

Future translation implementation:

One thing that you have pointed out, and something that will need further discussion, how will plugins in general handle translation.  This is something that we do need to discuss and implement at a later date.

**Comment by m1well** [comment]
@mikeerickson
I can't say anything about the new templating system or the new weather API, because I don't know about them.

but in my opinion there is a need to handle some translations (as written down in my json example) because you can't handle this
